torch/jit/__init__.py

```diff
@@ -1001,36 +1001,38 @@ def weighted_kernel_sum(self, weight):
     if not isinstance(inputs, dict):
         raise AttributeError("expected a dictionary of (method_name, input) pairs")
 
-    torch.jit._trace_module_map = {}
-
-    def register_submods(mod, prefix):
-        for name, child in mod.named_children():
-            submod_qualname = prefix + '.' + name
-            torch.jit._trace_module_map[child] = submod_qualname
-            register_submods(child, submod_qualname)
-
-    torch.jit._trace_module_map['__module'] = mod
-    register_submods(mod, '__module')
-
-    module = make_module(mod, _module_class, _compilation_unit)
-
-    for method_name, example_inputs in inputs.items():
-        # this is needed since Module.__call__ sets up some extra tracing
-        func = mod if method_name == "forward" else getattr(mod, method_name)
-        example_inputs = make_tuple(example_inputs)
-        module._c._create_method_from_trace(method_name, func, example_inputs, var_lookup_fn, _force_outplace)
-        check_trace_method = module._c._get_method(method_name)
-
-        # Check the trace against new traces created from user-specified inputs
-        if check_trace:
-            if check_inputs is not None:
-                _check_trace(check_inputs, func, check_trace_method,
-                             check_tolerance, _force_outplace, True, _module_class)
-            else:
-                _check_trace([inputs], func, check_trace_method,
-                             check_tolerance, _force_outplace, True, _module_class)
-
-    torch.jit._trace_module_map = None
+    old_module_map = torch.jit._trace_module_map
```

Comment:
can we add a comment here? So next time we don't have to "wtf" our way through the code again.

Reply:
I mean it's pretty self explanatory now. The issue was discovering that this was needed in the first place :p

Reply:
the usual pattern to do such things is to only replace the attribute when you finish computing it, i.e., start with a local `_trace_module_map` and only do `torch.jit._trace_module_map = _trace_module_map` at the end.